File: core/src/com/tiles/pathfinding/Player.java
package com.tiles.pathfinding;

public class Player {
    public boolean north;
    public boolean south;
    public boolean east;
    public boolean west;
    public boolean shortcutTaker;
    public boolean escalatorTaker;
    Player(boolean north, boolean south, boolean east, boolean west, boolean shortcutTaker, boolean escalatorTaker) {
        this.north = north;
        this.south = south;
        this.east = east;
        this.west = west;
        this.escalatorTaker = escalatorTaker;
        this.shortcutTaker = shortcutTaker;
    }

    public Player copyEscalator() {
        return new Player(north,south,east,west,shortcutTaker,false);
    }
    public Player copyShortcut() {
        return new Player(north,south,east,west,false,escalatorTaker);
    }
}

File: core/src/com/tiles/pathfinding/Player.kt
package com.tiles.pathfinding

import com.utils.Functions.modulo
import kotlinx.serialization.Serializable

@Serializable
class Player(
        val north: Boolean, val south: Boolean,
        val east: Boolean, val west: Boolean,
        val shortcutTaker: Boolean, val escalatorTaker: Boolean) {

    fun rotate(i: Int): Player =
            when (i) {
                0 -> this
                else -> Player(south, north, west, east, shortcutTaker, escalatorTaker).rotate(modulo(i - 1, 4))
            }

    /*
    WHAT
    THE
    FUCK
    THIS SHOULD NOT WORK LIKE THIS
    LET IT BE CLEAR: THIS RIGHT HERE IS WITCHCRAFT
    EVIL WITCHCRAFT
    DO NOT TOUCH THIS
    DO NOT REPRODUCE
    THIS CODE IS EVIL
     */

    /*
    Edit: This isn't witchcraft anymore
    At least I don't think it is ?
    It's just a classic inversion
    Thanks Mr. Simatic for the tip of using actual named constants
     */
}